pkg/upgrade/upgrade.go

```diff
@@ -487,6 +494,38 @@ func updateODCModelRegistry(ctx context.Context, cli client.Client, instanceName
 	return nil
 }
 
+// workaround for RHOAIENG-15328
+// TODO: this can be removed from ODH 2.22.
```

Comment:
what if user is directly upgrading from odh 2.20? can we ever really remove this?

Reply:
this fix will be included in ODH 2.21 (2.20 was released last week)
and we only support incremental upgrade in ODH,
user on 2.20 -> 2.21 (run into fix) then wont be problem again -> 2.22 , right?
we can keep it for 2.22 on the safe side, i only add this comments as a tech-debt so we do not keep the code there forever.